### PR TITLE
GS: Improve interlace handling and odd frames

### LINKED ISSUE #5262: [BUG]: NBA Jam (USA) - slus-20648 - f51b2328
### Describe the Bug

Stuck at the intro

### Reproduction Steps

Launch the game

### Expected Behavior

_No response_

### PCSX2 Revision

1.7.2190

### Operating System

Windows 10 (64bit)

### If Linux - Specify Distro

_No response_

### CPU

Ryzen 5 3600

### GPU

RTX 3070

### GS Settings

_No response_

### Emulation Settings

_No response_

### GS Window Screenshots

_No response_

### Logs & Dumps

[NBA Jam (USA) (2022-00-04 11-20-39).zip](https://github.com/PCSX2/pcsx2/files/7806898/NBA.Jam.USA.2022-00-04.11-20-39.zip)

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Attempt to improve interlace handling on games which were shaking or relying on the interlacing to display the picture (NBA Jam 2004)

### Rationale behind Changes
The frame/interlace handling to try and get the correct FIELD values reported and used.

### Suggested Testing Steps
test games with auto interlacing, make sure they don't shake or have half the picture missing or what have you.

Fixes #5262

### CODE REVIEW

**Comment by RedDevilus** [comment]
Potential games to check:

1. Mark of Kri
2. Urban Reign
3. Bakugan Battle Brawers (FMV)
4. I'll fill the others later unless someone else has shaky screen on defaults

**Comment by refractionpcsx2** [comment]
Mark of Kri seems fine except when it's loading/writing memcards, which is normal for a lot of PS2 games.
Bakugan seems good.
Don't have Urban Reign to hand.

**Comment by andercard0** [comment]
Urban Reign is working good here with this PR, tested 'til mission 3 and no more shakes... only thing I've noticed, and I do not know if that matters is when perform the last hit the screen shows the pixels as follows:

image

**Comment by RedDevilus** [comment]
Looks to be a texture cache issue which is beyond the scope of this PR.

**Comment by andercard0** [comment]
> Looks to be a texture cache issue which is beyond the scope of this PR.

Not a problem! I can live with it, important is what this PR covers, Thanks!

**Comment by JordanTheToaster** [comment]
Fixes the flickering on automatic in Conflict - Desert Storm II.

File: pcsx2/GS.cpp [hunk 1]
```diff
@@ -84,9 +84,10 @@ static __fi void gsCSRwrite( const tGS_CSR& csr )
 		//gifUnit.Reset(true); // Don't think gif should be reset...
 		gifUnit.gsSIGNAL.queued = false;
 		GetMTGS().SendSimplePacket(GS_RINGTYPE_RESET, 0, 0, 0);
-
+		u32 field = CSRreg.FIELD;
```
Comment (lightningterror) [inline pcsx2/GS.cpp:87]: Can be const.

**Comment by FuxorLuck** [comment]
Forever Kingdom and Echo Night beyond, both have half the picture missing, It is easily noticeable by moving the camera.

**Comment by refractionpcsx2** [comment]
is that the same on master? did you try changing the "Half screen fix" hw hack?

**Comment by FuxorLuck** [comment]
> is that the same on master? did you try changing the "Half screen fix" hw hack?

Yes, and Yes.

I am pretty sure it is an interlacing issue since I created no interlacing hack in order to fix the half screen. 
You can see the hack and the photos here

https://forums.pcsx2.net/Thread-No-interlacing-codes?pid=612598#pid612598

**Comment by FuxorLuck** [comment]
> > is that the same on master? did you try changing the "Half screen fix" hw hack?
> 
> Yes, and Yes.
> 
> I am pretty sure it is an interlacing issue since I created no interlacing hack in order to fix the half screen missing issue. You can see the hack and the photos here
> 
> https://forums.pcsx2.net/Thread-No-interlacing-codes?pid=612598#pid612598

**Comment by refractionpcsx2** [comment]
That looks to me more like a post processing effect that doesn't work properly, the game is trying to blur the stuff that's moving fast, software mode gets rid of the combing half screen effect.


Anyway, I think that's outside the scope of this PR and not really interlacing but something else going on.  The main aim of this PR was to resolve missing graphics (like NBA) and get rid of vertical shaking caused by deinterlacing not detecting correctly.

**Comment by Dreadmoth** [comment]
Testing _Transformers_ (SLES-52388):

60 Hz mode:
Master automatically uses "none" interlacing. The optimal mode seems to be blend **tff.** Blend bff is blurry in motion. Weave tff looks better than weave bff.
PR automatically uses blend tff. The optimal mode seems to be blend **bff**. Blend tff is blurry in motion. Weave bff looks better than weave tff.

50 Hz mode:
Master automatically uses blend tff, is good.
PR automatically uses blend bff, is good.

Bob modes shake and have combing artifacts on master and PR, 50 Hz and 60Hz modes.

**Comment by refractionpcsx2** [comment]
yep, that's what I expect :) The point of the PR is to correctly choose an interlace mode which doesn't judder up and down, unfortunately that does mean having a blurry interlace on some games, but I've tried to make it use bob where it can, as long as it doesn't cause shaking.

**Review by Mrlinkwii — APPROVED** [review]

**Comment by gtgamer468** [comment]
The PR still doesn't fixes the issue #1413.

**Comment by refractionpcsx2** [comment]
Yep, I didn't expect it to, but I guess I can look in to that.

**Review by RedDevilus — APPROVED** [review]
The only game that is still an issue (but lessened by a lot) is Urban Reign NTSC but it could be that timing issue that refraction told me about. But the other games that used to shake, don't anymore. So I'm fine with PR.

**Review by Mrlinkwii — APPROVED** [review]

**Pull request merged**